Adding the plugin to a Vite project with AdonisJS and Inertia cause an error

### Environment

- Operating System: Darwin
- Node Version:     v22.11.0
- Nuxt Version:     -
- CLI Version:      3.15.0
- Nitro Version:    -
- Package Manager:  npm@10.9.0
- Builder:          -
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -


### Version

Commit [845f85a](https://github.com/nuxt/ui/commit/845f85a072598f47c7afe10c4e5ebcc480450113)

### Reproduction

After creating a fresh AdonisJS project with the [Inertia and Vue 3 starter kit](https://docs.adonisjs.com/guides/getting-started/installation#inertia-starter-kit) 
```
# Vue without server-side rendering
npm init adonisjs@latest -- -K=inertia --adapter=vue --no-ssr
```

Here is the procedure followed to add NuxtUI to the project https://ui3.nuxt.dev/getting-started/installation/vue
1. Install the package with the latest commit, as the release has not yet been made 
```bash
npm install https://pkg.pr.new/@nuxt/ui@5a9511f
```

2. Add the Nuxt UI Vite plugin in the vite.config.ts
```typescript
import ui from '@nuxt/ui/vite'
import { defineConfig } from 'vite'
import { getDirname } from '@adonisjs/core/helpers'
import inertia from '@adonisjs/inertia/client'
import vue from '@vitejs/plugin-vue'
import adonisjs from '@adonisjs/vite/client'

export default defineConfig({
  plugins: [
    inertia({ ssr: { enabled: false } }),
    vue(),
    adonisjs({ entrypoints: ['inertia/app/app.ts'], reload: ['resources/views/**/*.edge'] }),
    ui(),
  ],

  resolve: {
    alias: {
      '~/': `${getDirname(import.meta.url)}/inertia/`,
    },
  },
})
```

Here I already have a problem: I can't launch the application.
```bash
failed to load config from /Users/kamil/Documents/projects/myproject/vite.config.ts
TypeError: The URL must be of scheme file
```

There is no problem with the use of the library with a Vite et Vue project alone.

It seems to be a problem with Vite or AdonisJS, but the error is not very explicit. I tried with Node 18, 20 and 22.

Here is the Replit to reproduce
https://replit.com/@kamilcaglarcont/Adonis-x-Inertia-x-Vue-3-x-Nuxt-UI?v=1

### Description

Hi, a few days weeks ago, the Nuxt UI 3 team made the component library [compatible with standalone Vue projects](https://github.com/nuxt/ui/issues/2129). Thank you for your work on this library 👏 !

I'd like to use it with AdonisJS and Inertia.

[AdonisJS](https://adonisjs.com) is a TypeScript-first web framework for building web apps and API servers. It is possible to use it with [Inertia](https://inertiajs.com) to build single-page apps with VueJS.

However, I've never managed to use them together, despite several hours of debugging. I have an error that seems to be related to Vite.

### Additional context

I tried with node 18, 20 and 22.

### Logs

```
[ info ] starting HTTP server...
Debugger attached.
failed to load config from /Users/kamil/Documents/projects/myproject/vite.config.ts

   TypeError: The URL must be of scheme file
```
No more logs even with `DEBUG=vite:* `

> [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/36273997?s=64&amp;u=cd8880fcceed9670aafb1bf6ddc6b085547234a1&amp;v=4">@kamilcglr](https://github.com/kamilcglr?rgh-link-date=2024-12-16T02%3A46%3A16.000Z), [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/1270155?s=64&amp;v=4">@UnluckyNinja](https://github.com/UnluckyNinja?rgh-link-date=2024-12-16T02%3A46%3A16.000Z) can u check if it works with this [#2899](https://github.com/nuxt/ui/pull/2899)?
> 
> ```
> npm i https://pkg.pr.new/@nuxt/ui@2899
> ```

Yes, it works for dev mode now. But when I tried to build it, it outputs a lot of type-checking errors emitted from the installed dep folder of this pr. (~260 errors, and most of them are almost identical so I picked only part of it)

```
node_modules/.pnpm/@nuxt+ui@https+++pkg.pr.new+@nuxt+ui@2899_@babel+parser@7.26.3_embla-carousel@8.5.1_radix-vue_4lbplemevlbktr25mrjxlvxfwu/node_modules/@nuxt/ui/dist/runtime/components/DropdownMenuContent.vue:35:30 - error TS2307: Cannot find module '#imports' or its corresponding type declarations.

35 import { useAppConfig } from '#imports'
                                ~~~~~~~~~~

node_modules/.pnpm/@nuxt+ui@https+++pkg.pr.new+@nuxt+ui@2899_@babel+parser@7.26.3_embla-carousel@8.5.1_radix-vue_4lbplemevlbktr25mrjxlvxfwu/node_modules/@nuxt/ui/dist/runtime/components/Form.vue:4:24 - error TS2307: Cannot find module '#build/app.config' or its corresponding type declarations.

4 import _appConfig from '#build/app.config'
                         ~~~~~~~~~~~~~~~~~~~

node_modules/.pnpm/@nuxt+ui@https+++pkg.pr.new+@nuxt+ui@2899_@babel+parser@7.26.3_embla-carousel@8.5.1_radix-vue_4lbplemevlbktr25mrjxlvxfwu/node_modules/@nuxt/ui/dist/runtime/components/Form.vue:166:41 - error TS2304: Cannot find name 'Event'.

166 async function onSubmitWrapper(payload: Event) {
                                            ~~~~~
```
The path error is indeed gone, thank you.

--- a/src/plugins/nuxt-environment.ts
+++ b/src/plugins/nuxt-environment.ts
@@ -1,5 +1,5 @@
 import type { UnpluginOptions } from 'unplugin'
-import { join, normalize } from 'pathe'
+import { normalize } from 'pathe'
 import { resolvePathSync } from 'mlly'
 import MagicString from 'magic-string'
 
@@ -9,7 +9,7 @@ import { runtimeDir } from '../unplugin'
  * This plugin normalises Nuxt environment (#imports) and `import.meta.client` within the Nuxt UI components.
  */
 export default function NuxtEnvironmentPlugin() {
-  const stubPath = resolvePathSync(join(runtimeDir, 'vue/stubs'), { extensions: ['.ts', '.mjs', '.js'] })
+  const stubPath = resolvePathSync('../runtime/vue/stubs', { extensions: ['.ts', '.mjs', '.js'], url: import.meta.url })
 
   return {
     name: 'nuxt:ui',

--- a/src/plugins/plugins.ts
+++ b/src/plugins/plugins.ts
@@ -15,9 +15,9 @@ import type { UnpluginOptions } from 'unplugin'
 export default function PluginsPlugin(options: NuxtUIOptions) {
   const plugins = globSync(['**/*', '!*.d.ts'], { cwd: join(runtimeDir, 'plugins'), absolute: true })
 
-  plugins.unshift(resolvePathSync(join(runtimeDir, 'vue/plugins/head'), { extensions: ['.ts', '.mjs', '.js'] }))
+  plugins.unshift(resolvePathSync('../runtime/vue/plugins/head', { extensions: ['.ts', '.mjs', '.js'], url: import.meta.url }))
   if (options.colorMode) {
-    plugins.push(resolvePathSync(join(runtimeDir, 'vue/plugins/color-mode'), { extensions: ['.ts', '.mjs', '.js'] }))
+    plugins.push(resolvePathSync('../runtime/vue/plugins/color-mode', { extensions: ['.ts', '.mjs', '.js'], url: import.meta.url }))
   }
 
   return {

--- a/src/unplugin.ts
+++ b/src/unplugin.ts
@@ -1,6 +1,6 @@
 import { fileURLToPath } from 'node:url'
 
-import { join } from 'pathe'
+import { join, normalize } from 'pathe'
 import { createUnplugin } from 'unplugin'
 import AutoImport from 'unplugin-auto-import'
 import { defu } from 'defu'
@@ -41,7 +41,7 @@ export interface NuxtUIOptions extends Omit<ModuleOptions, 'fonts' | 'colorMode'
   colorMode?: boolean
 }
 
-export const runtimeDir = fileURLToPath(new URL('./runtime', import.meta.url))
+export const runtimeDir = normalize(fileURLToPath(new URL('./runtime', import.meta.url)))
 
 export const NuxtUIPlugin = createUnplugin<NuxtUIOptions | undefined>((_options = {}, meta) => {
   const options = defu(_options, { fonts: false, devtools: { enabled: false } }, defaultOptions)

